trace/spans/factory.go

```diff
@@ -0,0 +1,341 @@
+package spans
```

Comment:
Are you planning on using OTEL at all? https://opentelemetry.io/docs/instrumentation/go/getting-started/

Reply:
> So I wanted a simple alternative to the typical tracing libraries, because Lager was already just needing a small nudge to be able to provide tracing

Gotcha, makes sense!

Reply:
Purely an out of interest discussion (non-blocking for this PR):

> but I don't see it integrating into a logging package like Lager

What about some of the things like Span lifecycle API, or even the propagation API to infer span/trace details from incoming headers? Such as inferring the span context, adding attributes, ending/starting a span etc? I see you have some of those interfaces defined for upcoming PRs?

Reply:
Yeah, one of the things I'm curious about is how much putting attributes on traces matters with GCP tracing when it is tied in with the logs and the logs have those details.  I wanted to experiment with that aspect.

Propagation is already handled by Lager.  The piece being added is having Lager create new spans rather than just propagating the one span that the GCP LB creates.  This includes simple lifespan management that dovetails with Lager's access logs.

So, Part I is spans.Factory which defines an interface for working with spans w/o having to swallow the GCP API library.  Part II is updating Lager to use this to simplify its existing Trace support and extend it to creating spans.  Part III is the currently empty 'trace' module which will implement a full-blown Factory that can create and modify spans and register them with GCP.

Part II and III are code complete and I'm just in the process of creates MRs for those.

Reply:
Thanks for the context, makes a lot of sense!

> The simplest version of adding trace to a server will be

Very nice, yeah that's really clean!